Rubberduck.Main/ComClientLibrary/UnitTesting/Fakes/Environ.cs

```diff
@@ -28,29 +28,28 @@ public override bool PassThrough
 
         [UnmanagedFunctionPointer(CallingConvention.StdCall, SetLastError = true)]
         [return: MarshalAs(UnmanagedType.BStr)]
-        private delegate string EnvironStringDelegate(IntPtr envstring, IntPtr number);
+        private delegate string EnvironStringDelegate(IntPtr envstring);
 
-        public string EnvironStringCallback(IntPtr envstring, IntPtr number)
+        public string EnvironStringCallback(IntPtr envstring)
         {
-            TrackInvocation(envstring, number);              
+            TrackInvocation(envstring);
             return ReturnValue?.ToString() ?? string.Empty;
         }
 
         [UnmanagedFunctionPointer(CallingConvention.StdCall, SetLastError = true)]
-        private delegate string EnvironVariantDelegate(IntPtr envstring, IntPtr number);
+        private delegate void EnvironVariantDelegate(IntPtr retVal, IntPtr envstring);
 
-        public object EnvironVariantCallback(IntPtr envstring, IntPtr number)
+        public void EnvironVariantCallback(IntPtr retVal, IntPtr envstring)
         {
-            TrackInvocation(envstring, number);
-            return ReturnValue?.ToString() ?? string.Empty;
```

Comment:
Going from memory and reviewing on a phone so pardon the possible nonsense, but wasn't `number` meant to track the invocation index? i.e. allowing to configure invocation 1 to do one thing, and invocation 2 to do another thing?

Reply:
I took those two to be based on the fact that Environ can take either a number or a string. However, there should only ever be one argument, it isn't two optional arguments. I tested that passing in a number worked as well as a string (although I didn't keep that integration test).

The invocation index would just be something you pass in with the .Returns or .ReturnsWhen calls but I will update the integration test and do a new commit to confirm.